src/org/ods/orchestration/util/DocumentHistory.groovy

```diff
@@ -0,0 +1,332 @@
+package org.ods.orchestration.util
+
+import com.cloudbees.groovy.cps.NonCPS
+import org.ods.orchestration.service.leva.ProjectDataBitbucketRepository
+import org.ods.orchestration.util.Project.JiraDataItem
+import org.ods.util.ILogger
+import org.ods.util.IPipelineSteps
+
+/**
+ * This class contains the logic for keeping a consistent document version.
+ */
+class DocumentHistory {
+
+    static final String ADD = 'add'
+    static final String ADDED = ADD + 'ed'
+    static final String DELETE = 'discontinue'
+    static final String DELETED = DELETE + 'd'
+    static final String CHANGE = 'change'
+    static final String CHANGED = CHANGE + 'd'
+
+    protected IPipelineSteps steps
+    protected ILogger logger
+    protected List<DocumentHistoryEntry> data = []
+    protected String targetEnvironment
+    /**
+     * Autoincremental number containing the internal version Id of the document
+     */
+    protected Long latestVersionId
+    protected final String documentType
+    protected Boolean allIssuesAreValid = true
```

Comment:
to improve encapsulation the following internal class members could be set to private

Reply:
As a comment, I have seen that groovy ignores private/protected tags. You can access anything from outside

Reply:
Groovy treats visibility keywords as a hint for a programmer. I think as a good practice to use this modifers for readability reasons. 
Maybe we have to define some groovy-coding-style guide... think about:
- https://dzone.com/articles/groovy-coding-style  
- https://google.github.io/styleguide/javaguide.html
- https://groovy-lang.org/style-guide.html